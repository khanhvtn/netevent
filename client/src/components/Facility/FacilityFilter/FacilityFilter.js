import React from 'react';
import {
    MuiPickersUtilsProvider,
    KeyboardDatePicker,
} from '@material-ui/pickers';
import {
    Grid,
    Typography,
    Button,
    Select,
    MenuItem,
    Drawer,
    InputLabel,
    FormControl,
} from '@material-ui/core';
import MomentUtils from '@date-io/moment';
//import useStyles in the last
import useStyles from './styles';

const FacilityFilter = ({
    openFilter,
    handleToggleFilter,
    statusFilter,
    handleFilterChange,
    createdFrom,
    createdTo,
    setFilters,
    updatedFrom,
    updatedTo,
    handleClearFilter,
    handleApplyFilter,
}) => {
    const css = useStyles();
    return (
        <Drawer anchor="right" open={openFilter} onClose={handleToggleFilter}>
            <div className={css.filterTitle}>
                <Typography style={{ fontWeight: 'bold' }} align="center" variant="h6">Filter Facility</Typography>
            </div>
            <div className={css.filterWrapper}>
                <div className={css.filterInputs}>
                    <FormControl fullWidth variant="standard">
                        <InputLabel id="statusFilterLabel">Status</InputLabel>
                        <Select
                            labelId="statusFilterLabel"
                            id="statusFiler"
                            value={statusFilter}
                            onChange={handleFilterChange}
                            label="Status"
                            inputProps={{
                                name: 'statusFilter',
                            }}
                        >
                            <MenuItem value="">
                                <em>None</em>
                            </MenuItem>
                            <MenuItem value={true}>Active</MenuItem>
                            <MenuItem value={false}>Expired</MenuItem>
                        </Select>
                    </FormControl>
                    <MuiPickersUtilsProvider utils={MomentUtils}>
                        <Grid container justify="space-around">
                            <KeyboardDatePicker
                                size="small"
                                fullWidth
                                margin="normal"
                                id="createdFrom"
                                label="Created From"
                                format="MM/DD/YYYY"
                                value={createdFrom}
                                onChange={(date) => {
                                    setFilters((prevState) => ({
                                        ...prevState,
                                        createdFrom: date.toDate(),
                                    }));
                                }}
                                KeyboardButtonProps={{
                                    'aria-label': 'change date',
                                }}
                            />
                            <KeyboardDatePicker
                                size="small"
                                fullWidth
                                margin="normal"
                                id="createdTo"
                                label="Created To"
                                format="MM/DD/YYYY"
                                value={createdTo}
                                onChange={(date) => {
                                    setFilters((prevState) => ({
                                        ...prevState,
                                        createdTo: date.toDate(),
                                    }));
                                }}
                                KeyboardButtonProps={{
                                    'aria-label': 'change date',
                                }}
                            />
                            <KeyboardDatePicker
                                size="small"
                                fullWidth
                                margin="normal"
                                id="updatedFrom"
                                label="Updated From"
                                format="MM/DD/YYYY"
                                value={updatedFrom}
                                onChange={(date) => {
                                    setFilters((prevState) => ({
                                        ...prevState,
                                        updatedFrom: date.toDate(),
                                    }));
                                }}
                                KeyboardButtonProps={{
                                    'aria-label': 'change date',
                                }}
                            />
                            <KeyboardDatePicker
                                size="small"
                                fullWidth
                                margin="normal"
                                id="updatedTo"
                                label="Updated To"
                                format="MM/DD/YYYY"
                                value={updatedTo}
                                onChange={(date) => {
                                    setFilters((prevState) => ({
                                        ...prevState,
                                        updatedTo: date.toDate(),
                                    }));
                                }}
                                KeyboardButtonProps={{
                                    'aria-label': 'change date',
                                }}
                            />
                        </Grid>
                    </MuiPickersUtilsProvider>
                </div>
                <div className={css.filterActions}>
                    <Button
                        className={css.handleClearButton}
                        onClick={handleClearFilter}
                        style={{ backgroundColor: 'transparent' }}
                        color="default"
                    >
                        Clear all
                    </Button>
                    <Button
                        className={css.handleFilterButton}
                        onClick={handleApplyFilter}
                        variant="contained"
                        color="primary"
                    >
                        Apply
                    </Button>
                </div>
            </div>
        </Drawer>
    );
};

export default FacilityFilter;
