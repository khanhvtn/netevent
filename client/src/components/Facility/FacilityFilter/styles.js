import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(() => ({
    filterActions: {
        marginTop: '24px',
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center',
    },
    filterInputs: {
        marginTop: '20px',
    },
    filterWrapper: {
        width: '250px',
        padding: '20px',
    },
    handleFilterButton: {
        textTransform: "none"
    },
    handleClearButton:{
        textTransform: "none",
        textDecoration: 'underline',
    }
}));
