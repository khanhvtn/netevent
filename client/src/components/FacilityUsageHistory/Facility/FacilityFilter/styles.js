import { makeStyles } from '@material-ui/core/styles';

export default makeStyles((theme) => ({
    filterActions: {
        marginTop: '24px',
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center',
    },
    filterInputs: {
        marginTop: '20px',
    },
    filterWrapper: {
        width: '250px',
        padding: '20px',
    },
    handleFilterButton: {
        textTransform: "none"
    },
    handleClearButton: {
        textTransform: "none",
        textDecoration: 'underline',
    },
    filterTitle: {
        padding: '16px 0',
        backgroundColor: theme.palette.background.default
    }
}));