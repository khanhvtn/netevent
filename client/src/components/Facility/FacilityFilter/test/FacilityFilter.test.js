import React from 'react'
import { create } from 'react-test-renderer'
import expect from 'expect'

import FacilityFilter from '../FacilityFilter'

describe('FacilityFilter Component', () => {
    test('Matches the snapshot', () => {
        const facilityFilter = create(<FacilityFilter />)
        expect(facilityFilter.toJSON()).toMatchSnapshot()
    });
});