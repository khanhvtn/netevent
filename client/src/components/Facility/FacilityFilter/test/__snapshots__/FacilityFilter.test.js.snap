// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacilityFilter Component Matches the snapshot 1`] = `null`;
